android/engine/src/main/java/org/smartregister/fhircore/engine/ui/pin/PinLoginScreen.kt

```diff
@@ -0,0 +1,270 @@
+  )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun PinLoginPage(
+  modifier: Modifier = Modifier,
+  onPinChanged: (String) -> Unit,
+  showError: Boolean = false,
+  onMenuLoginClicked: () -> Unit,
+  enterUserPinMessage: String = "",
+  forgotPin: () -> Unit,
+  appName: String = ""
+) {
+
+  var showMenu by remember { mutableStateOf(false) }
+  var showForgotPinDialog by remember { mutableStateOf(false) }
+
+  Surface(color = colorResource(id = R.color.white_slightly_opaque)) {
+    TopAppBar(
+      title = { Text(text = "", Modifier.testTag(PIN_TOOLBAR_TITLE)) },
+      navigationIcon = {
+        IconButton(onClick = {}) {
+          Icon(
+            Icons.Filled.ArrowBack,
+            contentDescription = "Back arrow",
+            modifier = Modifier.size(0.dp).testTag(PIN_TOOLBAR_MENU_ICON)
+          )
+        }
+      },
+      actions = {
+        IconButton(
+          onClick = { showMenu = !showMenu },
+          modifier = Modifier.testTag(PIN_TOOLBAR_MENU_BUTTON)
+        ) { Icon(imageVector = Icons.Outlined.MoreVert, contentDescription = null) }
+        DropdownMenu(
+          expanded = showMenu,
+          onDismissRequest = { showMenu = false },
+          Modifier.testTag(PIN_TOOLBAR_MENU)
+        ) {
+          DropdownMenuItem(
+            onClick = {
+              showMenu = false
+              onMenuLoginClicked()
+            },
+            modifier = Modifier.testTag(PIN_TOOLBAR_MENU_LOGIN)
+          ) { Text(text = stringResource(id = R.string.pin_menu_login)) }
+        }
+      }
+    )
+
+    if (showForgotPinDialog) {
+      ForgotPinDialog(forgotPin = forgotPin, onDismissDialog = { showForgotPinDialog = false })
+    }
+
+    Column(
+      modifier =
+        Modifier.fillMaxSize()
+          .padding(horizontal = 16.dp, vertical = 70.dp)
+          .wrapContentWidth(Alignment.CenterHorizontally)
+    ) {
+      val appLogo =
+        when (appName) {
+          "g6pd" -> R.drawable.ic_logo_g6pd
+          else -> R.drawable.ic_liberia
+        }
+      Image(
+        painter = painterResource(id = appLogo),
+        contentDescription = stringResource(id = R.string.app_logo),
+        modifier =
+          modifier
+            .padding(top = 16.dp)
+            .align(Alignment.CenterHorizontally)
+            .requiredHeight(120.dp)
+            .requiredWidth(140.dp)
+            .testTag(APP_LOGO_TAG),
+      )
+      Text(
+        text = appName,
+        textAlign = TextAlign.Center,
+        fontWeight = FontWeight.Bold,
+        fontSize = 22.sp,
+        modifier = modifier.padding(top = 16.dp).align(Alignment.CenterHorizontally)
+      )
+
+      Text(
+        text = enterUserPinMessage,
+        textAlign = TextAlign.Center,
+        fontWeight = FontWeight.Normal,
+        fontSize = 20.sp,
+        modifier = modifier.padding(vertical = 16.dp).align(Alignment.CenterHorizontally)
+      )
+
+      PinView(
+        pinInputLength = 4,
```

Comment:
this value should come from the same constant or config as https://github.com/opensrp/fhircore/pull/1030/files#diff-d63f836c3cd32ded90b5d79304289b00d7db17d9b2bb9d06f0a8e1f62156dc0eR73

Reply:
fixed